new_device.py

```diff
@@ -0,0 +1,171 @@
+import torch
+from torch.overrides import enable_torch_function_mode, TorchFunctionMode
+from torch.utils._pytree import tree_map
+
+import numpy as np
+
+aten = torch.ops.aten
+
+# 1. A Tensor that stores custom raw_data and implement functions for it
+class MyDeviceTensor(torch.Tensor):
+    IMPLEMENTATIONS = {}
+
+    @staticmethod
+    def __new__(cls, size, dtype, raw_data=None, requires_grad=False):
+        # Use a meta Tensor here to be used as the wrapper
+        return torch.Tensor._make_subclass(cls, torch.empty(size, dtype=dtype, device="meta"), require_grad=requires_grad)
+
+    def __init__(self, size, dtype, raw_data=None, requires_grad=False):
+        # Store any provided user raw_data
+        self.raw_data = raw_data
+
+    def __repr__(self):
+        st = super().__repr__()
+        st = st.replace("device='meta'", "device='my_device'")
+        # Print the content the best way possible
+        new_content = "[" + ", ".join(str(el) for el in self.raw_data) + "]"
+        st = st.replace("...", new_content)
+        return st
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        if func in cls.IMPLEMENTATIONS:
+            try:
+                def super_fn(*args, **kwargs):
+                    return super(cls, MyDeviceTensor).__torch_dispatch__(func, types, args, kwargs)
+                return cls.IMPLEMENTATIONS[func](super_fn, *args, **kwargs or {})
+            except Exception as e:
+                print(e)
+                raise e
+        raise RuntimeError(f"No implementation for 'my_device' for {func}, {args}, {kwargs}")
```

Comment:
If you use some of the tech from @eellison's fake tensor you should also be able to get the device property to report correctly as well

Reply:
I haven't looked into that one yet.
The problem is that even if that tries to create a `torch.device` based on the result from c++, that will fail.